ipa-core/src/hpke/mod.rs

```diff
@@ -231,9 +221,12 @@ mod tests {
             match_key.serialize(&mut plaintext);
 
             let (encap_key, ciphertext, tag) = seal_in_place(
-                &self.registry,
+                self.registry
+                    .public_key(info.key_id)
+                    .ok_or(CryptError::NoSuchKey(info.key_id))
+                    .unwrap(),
```

Comment:
this requires changing the contract of this function and updating callers. Given that encryption is not used inside MPC, I am less worried about this going wrong. Although it may be nice to do this work eventually